### DISCUSSION TITLE
The validation rules are not triggered correctly when we declare an array of fields and not the fields themselves

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#7704] [4. General]
Hi, 

I am dealing with the metadata of images. There are 20-30 fields in the EXIF properties of images. 

All the properties and their rules are written in an associative array `$data` in a config file.

Can I declare all the fields and attach a `#[Validate]` to them directly without having to re-write all the fields one by one? We notice that this code syncs the field `key` inside the array `$ArrayOfKeys` but doesn't read the rule attached. 

The class
```PHP
#[Validate]
public $arrayOfKeys = [];

#[Validate]
public $key;

public function rules()
    {
        return [
            'arrayOfKeys.key' => 'required|min:3',
            'key' => 'required|min:3',
        ];
    }
```

And the view 
```PHP
{{var_dump($errors)}}
<br>
KEY
<input type="text" label="key" value="" wire:model.live.debounce="key"/>
<br>
TEST.KEY
<input type="text" value="" wire:model.live.debounce="arrayOfKeys.key"/>
```

The `$errors` receives the correct errors and is triggered correctly for the KEY field as we type.
The rules are not applied correctly to the `ArrayOfKeys.key` and no error is received. However, the value of ArrayOfKeys.key is sync in the view (if we declare it), and the rules() method is triggered as we type.

Have you an idea ? Thank you :pray:

**Comment by joffreypersia** [comment]
Up Please
If you need more information, I can do my best to provide more details.
Thank you !

**Comment by joffreypersia** [comment]
Up..?

**Comment by LaaHudra** [comment]
Same Issue here

**Comment by LaaHudra** [comment]
Hey @joffreypersia:

I convert nested array to Form Class and works like a charm.
https://livewire.laravel.com/docs/forms#extracting-a-form-object

**Reply by joffreypersia** [reply]
Hi @LaaHudra 

Thank you for your help.

The problem I am facing with my colleague is that we need to declare every variable field one by one if we want to associate them with an array of rules.

I would like to declare the fields from an array of a config file and associate them with rules from an array too.

We also use the LiveWire form object. Does it work in your project ?

Thank you

**Reply by Rikaelus** [reply]
This works in an ideal world but not in a world of child components with a limit of only one Modelable property. The only way I know of to have the child component deal with multiple data points is to have them in a nested array -- in my case 3 levels deep.

**Comment by Rikaelus** [comment]
I'm running into this same issue in a situation where I'm even more constrained. Not only am I dealing with dynamic lists of variables to collect values for but I'm working within a child component that can have only a lone `#[Modelable]` property coming in, having to store a decent amount of structured data.

So I'm really feeling this pain and it definitely feels like a bug. The fact you can make realtime updates (ie., `.blur`, et al.) but not have them trigger validation as a conventional property would... can't be right.

**Reply by lrljoe** [reply]
With a child component. validation should be done within the context of that component.

The child component per Modelable should only be returning a single value.

Otherwise you're probably misusing child components!

**Reply by Rikaelus** [reply]
@lrljoe Misusing child components based on what? The documentation goes on at length about how components are components with each being an "island" of independent functionality, regardless of relation to each other. And components are defined as being used to encapsulate reusable UI elements which is exactly what I'm doing. The documentation even states Livewire provides the `#[Modelable]` attribute that "you can add to **any** child component property" -- it doesn't cite any limitation to single-value properties and, indeed, it works fine if you use it on an array _except_ where it comes to validation as is being reported here.

tl;dr: the documentation never frames child components as being inferior to parent components in functionality, nor Modelable as only being intended for single-value properties.
